hermes/apps/athena/modules/staked-ada-indexer/src/index.rs

```diff
@@ -0,0 +1,175 @@
+//! Collects staked TXO and TXI indexed data.
+//!
+//! Referenced from:
+//! - <https://github.com/input-output-hk/catalyst-voices/blob/9bb741c2637bf8fe6814a1dd4e3fe986de536d67/catalyst-gateway/bin/src/db/index/block/txo/mod.rs>
+//! - <https://github.com/input-output-hk/catalyst-voices/blob/9bb741c2637bf8fe6814a1dd4e3fe986de536d67/catalyst-gateway/bin/src/db/index/block/txi.rs>
+
+use shared::{
+    cardano_blockchain_types::{
+        hashes::{Blake2b256Hash, TransactionId},
+        pallas_addresses::{Address, ShelleyDelegationPart},
+        pallas_traverse::{MultiEraOutput, MultiEraTx},
+        MultiEraBlock, Network, StakeAddress,
+    },
+    database::staked_ada::{TxiByTxnIdRow, TxoAssetsByStakeRow, TxoByStakeRow},
+    utils::log::{error, warn},
+};
+
+/// Temporary buffers, holding entries to be inserted.
+#[derive(Default)]
+pub struct Buffers {
+    pub txo_by_stake: Vec<TxoByStakeRow>,
+    pub txo_assets_by_stake: Vec<TxoAssetsByStakeRow>,
+    pub txi_by_txn_id: Vec<TxiByTxnIdRow>,
+}
+
+impl Buffers {
+    /// Index all transaction inputs and outputs in a block.
+    pub fn index_block(
+        &mut self,
+        block: &MultiEraBlock,
+    ) {
+        for (txn_index, txn) in block.enumerate_txs() {
+            let txn_id = TransactionId::from(Blake2b256Hash::from(txn.hash()));
+            self.index_txn_txo(
+                block.network().clone(),
```

Comment:
We do not have to `.clone()` here, since the only thing we do with the cloned `network` is to clone it again in `index_txn_txo()`. So this could be a borrow.